src/rgw/rgw_acl.cc

```diff
@@ -166,21 +166,29 @@ uint32_t RGWAccessControlPolicy::get_perm(const DoutPrefixProvider* dpp,
   ldpp_dout(dpp, 20) << "-- Getting permissions begin with perm_mask=" << perm_mask
                  << dendl;
 
-  uint32_t perm = acl.get_perm(dpp, auth_identity, perm_mask);
-
-  if (auth_identity.is_owner_of(owner.id)) {
-    perm |= perm_mask & (RGW_PERM_READ_ACP | RGW_PERM_WRITE_ACP);
-  }
+  uint32_t perm = 0;
+  const uint32_t user_perm = auth_identity.get_perm_mask();
+  if (user_perm == RGW_PERM_FULL_CONTROL) {
+    // Only consider ACL if the user has full control (either a subuser with full control or a non-subuser)
+    perm = acl.get_perm(dpp, auth_identity, perm_mask);
+
+    if (auth_identity.is_owner_of(owner.id)) {
```

Comment:
Hi @cbodley, I was wondering if this needs to be changed for account implementation as the user can be part of the account but not necessarily a root user.
Basically, if I understand this correctly, this should be only applied to the root user (https://github.com/ceph/ceph/commit/2365c77aad8a034c6a7fae0fae8b393805f3ca28#diff-c4026f6b05b0b02d46a4d07c3b707f976a7a566691064b84d280b2d3f9c63882L69)